### PR TITLE
improving USB and 2FA

### PR DESCRIPTION
**Pull request by cottsak**
some of this didn't read very well:
* cleared up the 2FA section
* added TOTP which is the answer where SMS is not
* removed the section about the "yubikey" USB 2FA alternative - this is contradictory advice when you're giving great advice at the start regarding banning USB devices (see my additionally clarification there too). it's just too hard to teach folks that some USB are ok but most are not. So keep it simple and use TOTP and ban all USB devices period!

### CODE REVIEW

**Comment by itaifrenkel** [comment]
@cottsak The issue with TOTP is that it causes alot of friction (rebellion is a better word). Especially when you need it over and over again. That's why push notifications are used. 

Some employees would not harden their mobile phone, or would not use it for work purposes. What then? This is when you can introduce yubikeys. Device controls can be configured to allow ubikey and not storage devices.

I think that we should introduce all aspects of this, and let each company choose their culture fit.

**Comment by cottsak** [comment]
@itaifrenkel 

I'm sure yubikeys are a great 2FA option. The problem I have is with the conflicting advice in the document. If on one hand you want folks to steer clear of USB devices (which is awesome advice) then, for most companies, it makes sense to make that sweeping and avoid yubikeys. If you don't, then folks won't know where the line is. There will form a "grey area" where some USB devices are ok but others aren't. As humans, we don't handle those types of complicated constraints well and eventually the guidance will relax and everyone will be using all sorts of USB devices hence defeating any gain with the initial "avoid USB thumb drives" you had to begin with.

If on the other hand you think the 2FA protection from yubikeys outweighs the physical access risks associated with the "don't use any USB devices" then you'd simply be wrong. 2FA is a "hardening" measure on top of machine-based authentication. Physical access almost exclusively results in machine-root level access and will always be the bigger risk.

If I were you I would reconsider the yubikeys advice. Like my PR suggests: keep things simple for readers and drop the conversation about yubikeys altogether.

**Comment by cottsak** [comment]
@itaifrenkel Now coming back to TOTP

> The issue with TOTP is that it causes alot of friction

This sounds like personal experience more than fact.

TOTP codes are widely used and products like Google Authenticator aren't limited to the Google product offering - any application can use a TOTP code generator. I don't see what the "friction" or "rebellion" is. They are potentially not as smooth a UX as an sms with a code but if anything, it's a marginal cost. Once folks get used to the app they will be fine. Just like folks get used to pulling their RSA keyfob from their pocket before authenticating- it's the same. It's not a big deal and you will see more vendors swapping (or at least adding) to TOTP in the light of the recent [NIST advice asking vendors to move away from it](https://www.schneier.com/blog/archives/2016/08/nist_is_no_long.html).

**Comment by gvangool** [comment]
@cottsak the problem is more that users don't want to use personal smartphones for work products. They feel that if the company wants them to use a TOTP app, that the company needs to provide the phone to run it on.

**Comment by itaifrenkel** [comment]
@cottsak I added (even more) text to the 2FA section. It's not as elegant, but it delivers the different tradeoffs discussed here better (I think). WDYT? (CC @gvangool )

**Comment by itaifrenkel** [comment]
@winbuzz2523 would you care to elaborate what's inside the pdf file, before we open it :))

**Comment by cottsak** [comment]
@gvangool how is that different for an SMS? You still need a phone for that

File: security.md [hunk 1]
```diff
@@ -101,12 +101,15 @@ Using git would allow you to add outsource/freelance developers for a limited ti
...
+* A well established and easy to use 2FA implementation is [TOTP 2FA implimentation](https://en.wikipedia.org/wiki/Time-based_One-time_Password_Algorithm). Employees can install [Google Authenticator](https://en.wikipedia.org/wiki/Google_Authenticator) on their mobile phone and it would produce a one-time 6 digit code that needs to be typed in addition to the password 
 
+* A slightly more usable approach are push notifications. Many SaaS vendors support an app that instead of generating codes, sends you a push notification. That is more convinient than copying 6 digits each time you need to log-in. The downside is that you would probably need different apps for each service, and that it won't work without an internet connection, or when the push notification vendor has downtime.
```
Comment (cottsak) [inline security.md]: I've never seen or used one of these. Can you cite an example please?

**Comment by itaifrenkel** [comment]
@cottsak Most employees won't say anything about SMS, TOTP and even push notification. About ~20% of them would start to feel uncomfortable when you ask them to lock their phone and encrypt it , in order to protect those apps.

The other objects are ofcourse usability (the fact that you need to take out your phone unlock it and push/copy numbers)

File: security.md [hunk 2]
```diff
@@ -101,12 +101,15 @@ Using git would allow you to add outsource/freelance developers for a limited ti
...
+
+* Mobile phone 2FA problems start when an employee lost/forgot/replaces their mobile phone, or is stuck without battery, and then they are locked out of the system. Some employees don't like hardening their phone (encryption and screenlock to protect the 2FA app), and feel that the company needs to buy the phone for the 2FA app to run on. That's why some organizations use yubikey instead (a small usb plug that can sense human finger touch). Like a Mobile App Push notification, yubikey protects against malware and hackers that keylog your password. The downside is that unlike their mobile phones, employees are more likely to lose the yubikey together with their laptop. So if the stolen laptop falls into the hands of a professional that can extract the passwords from the laptop then they could also login to any service with the yubikey. Another problem is that the yubikey is a USB device, and you would need to make the distinction between USB storage devices (not allowed) and the yubikey (allowed).
```
Comment (cottsak) [inline security.md]: So this clarity is good.
I personally would drop it for brevity however. I'm not sure of the goals of this advice - to what level of detail do you aspire and who is the target audience?

File: security.md (hunk 1 above)
Comment (itaifrenkel) [inline security.md]: Look for "Okta Verify", "Lastpass Authenticator" or "Duo Push" or "Google Prompt"

Comment (cottsak) [inline security.md]: Novel idea. +1

**Comment by cottsak** [comment]
@itaifrenkel no problem. I think we've made some good changes here. I'd be good to merge it in.

**Pull request merged**